### PR TITLE
Auto-complete improvements

### PR DESCRIPTION
**Pull request by dinhani**
This PR improves the autocomplete features adding two functionalities:

1. Automatically adds the equals sign (` = `) after the property name when the line does not have one.

2. Changes the order some property values are displayed:
 * `true` always comes first
 * `false` always comes after `true`
 * `unset` is always the last value

Example:

editorconfig-vscode-autocomplete-improvements

### CODE REVIEW

File: src/EditorConfigCompletionProvider.ts [hunk 1]
```diff
@@ -213,7 +229,28 @@ class EditorConfigCompletionProvider implements CompletionItemProvider {
...
+						completionItem.sortText = '1'; // always first
+						break;
+					case 'false':
+						completionItem.sortText = '2'; // always second
+						break;
+					case 'unset':
+						completionItem.sortText = '9'; // always the last
+						break;
+					default:
+						completionItem.sortText = '3' + value; // after true or false, keeping the natural order
```
Comment (jedmao) [inline src/EditorConfigCompletionProvider.ts]: What is `+ value` achieving here?

File: src/EditorConfigCompletionProvider.ts [hunk 2]
```diff
@@ -213,7 +229,28 @@ class EditorConfigCompletionProvider implements CompletionItemProvider {
 		values: string[]
 	): CompletionItem[] {
 		return values.map(
-			value => new CompletionItem(value, CompletionItemKind.Value)
+			value => {
+				// basic info
+				const completionItem = new CompletionItem(value, CompletionItemKind.Value);
+
+				// sort predefined values to specific order
+				switch (value) {
```
Comment (jedmao) [inline src/EditorConfigCompletionProvider.ts]: Consider using a hash here like so:

```ts
completionItem.sortText = ({
    true: 1,
    false: 2,
    unset: 9
})[value] || '3' + value;
```

File: src/EditorConfigCompletionProvider.ts (hunk 1 above)
Comment (dinhani) [inline src/EditorConfigCompletionProvider.ts]: `+ value` has the purpose to keep the natural order.

For example, in `end_of_line`, values will be `3cr`, `3crlf` and `3lf`. If I don't append the value, all of them will have `3` as `sortText` and the order will probably break.

Comment (jedmao) [inline src/EditorConfigCompletionProvider.ts]: Cool strategy!

**Comment by jedmao** [comment]
Hey @dinhani. I just played with it and it's so awesome! Do you happen to know if there's a way to automatically pop-up Intellisense after the ` = ` is added? Right now, I have to hit `Ctrl+Space` to see it pop up. Just wondering.

**Comment by dinhani** [comment]
@jedmao 
There is a way, but it involves a workaroud.

First we need to create a custom command like this:

```typescript
commands.registerCommand('triggerSuggestAfterDelay',
	() => {
		setTimeout(function () {
			commands.executeCommand('editor.action.triggerSuggest');
		}, 100);
	}
);
```

Then we set the command in the `completionItem` item like this:

```typescript
completionItem.command = {
    arguments: [],
    title: '',
    command: 'triggerSuggestAfterDelay'
};
```

The workaround is the custom command with `setTimeout` we have to create because if we invoke `editor.action.triggerSuggest` directly, IntelliSense does not appears because the command is invoked before the text is inserted.

If you agree with this aproach, I can make a PR. 

I did not find any other way. Maybe working with events might work, like `TextDocumentContentChangeEvent`, but I think the custom command solution is simpler to work.

**Comment by jedmao** [comment]
That looks like a perfectly acceptable solution to me.

**Pull request merged**